lib/wibox/widget/calendar.lua

```diff
@@ -139,8 +139,23 @@ end
 -- @tparam number|nil date.day Date day
 -- @treturn widget Grid layout
 local function create_month(props, date)
-    local num_rows    = 8
-    local num_columns = props.week_numbers and 8 or 7
+    local start_row    = 3
+    local week_start   = props.start_sunday and 1 or 2
+    local last_day     = os.date("*t", os.time{year=date.year, month=date.month+1, day=0})
+    local month_days   = last_day.day
+    local column_fday  = (last_day.wday - month_days + 1 - week_start ) % 7
+
+    local num_columns  = props.week_numbers and 8 or 7
+    local start_column = num_columns - 6
+    local num_rows     = last_day.day // 7
```

Comment:
Integers and the integer division operator have only been introduced in Lua 5.3. Awesome supports Lua versions as old as 5.1, where this will result in a syntax error.

Reply:
Oh yeah, didn't think about that, I'll fix it.